saleor/graphql/core/scalars.py

```diff
@@ -34,14 +36,19 @@ def parse_value(value):
 class WeightScalar(graphene.Scalar):
     @staticmethod
     def parse_value(value):
-        # Expects value to be a string "amount unit" separated by a single
-        # space.
-        try:
-            value = decimal.Decimal(value)
-        except decimal.DecimalException:
-            return None
-        default_unit = get_default_weight_unit()
-        return Weight(**{default_unit: value})
+        weight = None
+        if isinstance(value, dict):
+            weight = Weight(**{value["unit"]: value["value"]})
+        else:
+            try:
+                value = decimal.Decimal(value)
+            except decimal.DecimalException:
+                return None
+            default_unit = get_default_weight_unit()
+            weight = Weight(**{default_unit: value})
```

Comment:
Can't we use here `parse_literal_decimal` method?

Reply:
Yes, we can

Reply:
Done